frontend/src/components/pages/HomePage.tsx

```diff
@@ -1,31 +1,172 @@
-import React, { useState } from "react";
-import { gql, useQuery } from "@apollo/client";
+import React, { useContext, useState } from "react";
+import { DocumentNode, gql, useQuery } from "@apollo/client";
 import StoryList from "../story/StoryList";
 import { StoryCardProps } from "../story/StoryCard";
+import { Role } from "../../constants/Enums";
+import AuthContext from "../../contexts/AuthContext";
 
-const GET_STORIES = gql`
-  query GetStories {
-    stories {
-      id
-      title
-      description
-      youtubeLink
-      level
-    }
-  }
-`;
+export enum HomePageDisplayState {
+  MyWork,
+  Browse,
+}
+
+type QueryInformation = {
+  fieldName: string;
+  string: DocumentNode;
+};
+
+type StoryResponseDTO = Omit<StoryCardProps, "cardId"> & {
+  // cardId will be assembled based on these fields
+  storyId: number;
+  storyTranslationId?: number;
+};
 
 const HomePage = () => {
+  // how do private methods exist?
+  const buildHomePageStoriesQuery = (
```

Comment:
> There's a bit of repetition that I kept in for clarity. It makes it very clear what the query is. Thoughts?

It seems that theres only 3 lines changing in the gql query strings themselves, eg:
```
storyTranslationsByUser(
            userId: ${userId},
            translator: ${role === Role.Translator}
          )
```
Perhaps the rest of the query string can be abstracted away and the returns in this function can be more succinct

>WHy does else if not work??? The linter doesn't like it. Should I be using switch-cases?

Is the error you're getting `Unnecessary 'else' after 'return'  no-else-return`? To avoid this, you can initialize a return variable and change it in the if statement
```
let res = {fieldName:"storyTranslationsByUser", string:... }
if (...){
  res = ....
} else if (...) {
  res = ....
}
return res;
```
> Where should these private-type methods exist? As in not used outside of component, not re-used -> it's only in its own method to clean up the reading for the bottom lines.

two thoughts: 
- Building query strings for repeat use is actually desirable - it encourages devs to have consistent expectations of DTO shape. It can also help us out in funky situations like the `refresh` mutation, where the user **must** request for a refresh_token. Something we should look into is storing and managing all our queries
- What you have going on here indeed scopes the function to `HomePage` - which a 'private function' in this setting. More about private functions in prototypes and classes [here](https://stackoverflow.com/questions/55611/javascript-private-methods?page=1&tab=votes#tab-top) and [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields). Note that we're likely to never explicitly write a prototype or class in this codebase - react is real f u n c t i o n a l these days.